pipeline/backend/ssh/ssh.go

```diff
@@ -0,0 +1,141 @@
+package ssh
+
+import (
+	"context"
+	"encoding/base64"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/melbahja/goph"
+
+	"github.com/woodpecker-ci/woodpecker/pipeline/backend/types"
+	"github.com/woodpecker-ci/woodpecker/server"
+)
+
+type ssh struct {
+	cmd    *goph.Cmd
+	output io.ReadCloser
+	client *goph.Client
+}
+
+type readCloser struct {
+	io.Reader
+}
+
+func (c readCloser) Close() error {
+	return nil
+}
+
+// make sure local implements Engine
+var _ types.Engine = &ssh{}
+
+// New returns a new ssh Engine.
+func New() types.Engine {
+	return &ssh{}
+}
+
+func (e *ssh) Name() string {
+	return "ssh"
+}
+
+func (e *ssh) IsAvailable() bool {
+	return true
+}
+
+func (e *ssh) Load() error {
+	var auth goph.Auth
+	if file, has := os.LookupEnv("WOODPECKER_SSH_KEY"); has {
+		var err error
+		auth, err = goph.Key(file, os.Getenv("WOODPECKER_SSH_KEY_PASSWORD"))
+		if err != nil {
+			return err
+		}
+	} else {
+		auth = goph.Password(os.Getenv("WOODPECKER_SSH_PASSWORD"))
```

Comment:
What about people using key + password as a form a poor-man 2 factor authentication ?
(that's supported since years: https://security.stackexchange.com/questions/17931/possible-to-use-both-private-key-and-password-authentication-for-ssh-login )

Reply:
I don't see any way to do this in goph or I didn't see it.

Reply:
I guess that's a rather niche use case. Maybe a upstream bug could be opened on it, and/or documented ?

Reply:
https://github.com/melbahja/goph/issues/36